pkg/entities/set.go

```diff
@@ -43,28 +41,36 @@ const (
 // Do not expose set to IPFIX library user
 // Not creating any interface. Plan to use same struct for Template and Data sets
 
-type Set struct {
+type Set interface {
+	AddRecord(elements []*InfoElement, templateID uint16, isDecoding bool)
+	GetRecords() []Record
+	GetNumberOfRecords() uint32
+}
+
+type BaseSet struct {
 	// Pointer to message buffer
 	buffer  *bytes.Buffer
 	currLen uint16
 	setType ContentType
 }
 
-// enterpriseID -> elementID
-type TemplateSet map[uint32][]uint16
+type TemplateSet struct {
+	records []*templateRecord
+}
 
-// enterpriseID -> elementID -> val
-type DataSet map[uint32]map[uint16]interface{}
+type DataSet struct {
+	records []*dataRecord
+}
 
-func NewSet(buffer *bytes.Buffer) *Set {
-	return &Set{
+func NewSet(buffer *bytes.Buffer) *BaseSet {
```

Comment:
is this used anywhere?
Set interface and other structures BaseSet, TemplateSet, DataSet seems confusing. Which structures are implementing Set interface? I do not see BaseSet embedded in TemplateSet or DataSet?